File: assignment_w2/hw4.py
def twoSum(nums, target):
    # 先將 target 拆成一半，減少時間
    # 舉例來說總和為 9 ，找 2 就和找 7 的意思一樣
    reduced = 0
    if target/2 == 0:
        reduced = int(target/2)
    else:
        reduced = int((target+1)/2)
    pool = list(range(reduced))

    idx = []
    for n in pool:
        if n in nums:
            idx1 = nums.index(n)
            idx2 = nums.index(target-n)
            idx.append(idx1)
            idx.append(idx2)
    return idx

result = twoSum([2,11,7,15], 9)
print(result) #[0,2]

File: assignment_w2/hw4.js
function twoSum(nums, target){
    var reduced = 0;
    if (target/2 == 0){
        reduced = target/2;
        console.log(reduced)
    } else{
        reduced = (target+1)/2;
    }
    var idx = [];
    for (i=0; i<reduced; i++){
        if (nums.includes(i)){
            var idx1 = nums.indexOf(i);
            var idx2 = nums.indexOf(target-i);
            if (idx1>=0 && idx2>=0){
                idx.push(idx1);
                idx.push(idx2);
            } //因為找不到會回傳 -1, 我們不要-1
        }
    }
    return idx;
}  

console.log(twoSum([2,11,7,15], 9)); // [0,2]